### ISSUE TITLE
Lambda functions to process changes to organizations, repositories and contacts

### ISSUE THREAD

**Issue by mfenner** [datacite/datacite#1209]

**Comment by paulvierkant** [comment]
- [ ] changed the role of a user in stage Fabrica and it synced correctly to stage SF
- [ ] created a new user in stage Fabrica (including new role) and it synced correctly to stage SF
- [ ] assigned roles of users in stage Fabrica so that one contact lost its role in stage Fabrica

**Comment by lizkrznarich** [comment]
Added role Secondary technical contact in Fabrica to existing contact which previously had no role. Role appears as Primary technical contact in Salesforce. 

Fabrica contact: https://doi.stage.datacite.org/contacts/7ec7c296-5d85-462b-a43e-20d8ca536dd6
Corresponding Salesforce contact: https://datacite--sync.lightning.force.com/lightning/r/Contact/0039E00001CWRwBQAX/view


<img width="828" alt="Screen Shot 2021-03-29 at 8 42 45 AM" src="https://user-images.githubusercontent.com/8026886/112849485-bdcc0980-906e-11eb-9f2e-7f2064771e84.png">
<img width="826" alt="Screen Shot 2021-03-29 at 9 09 05 AM" src="https://user-images.githubusercontent.com/8026886/112849502-c0c6fa00-906e-11eb-8f0f-20b81c3996e2.png">

**Comment by lizkrznarich** [comment]
Edited name of contact that existed in both Fabrica and Salesforce from Lizbert Jane to Lizabert Googlesnert
Fabrica contact: https://doi.stage.datacite.org/contacts/7ec7c296-5d85-462b-a43e-20d8ca536dd6
Salesforce contact: https://datacite--sync.lightning.force.com/lightning/r/Contact/0039E00001CWRwBQAX/view
- Name change was not reflected on organization settings in Fabrica
- Name change was not synced to Salesforce record (change made at 10:48 CDT, just before an hourly sync)

<img width="998" alt="Screen Shot 2021-03-29 at 10 48 19 AM" src="https://user-images.githubusercontent.com/8026886/112869533-e3630e00-9082-11eb-9094-041ba1651107.png">
<img width="985" alt="Screen Shot 2021-03-29 at 10 48 04 AM" src="https://user-images.githubusercontent.com/8026886/112869545-e6f69500-9082-11eb-9376-7aef5e3069e1.png">

**Comment by paulvierkant** [comment]
I let Meister Petz and Reinecke Fuchs both take Porky Pigs roles in Fabrica. However, Porky Pig (now renamed to Porky Vierkant, to let him end up on the report for the double opt-in mailing list), still has the role of the voting and service contact.
So the roles were not updated accordingly.

https://doi.stage.datacite.org/providers/ekrc
https://doi.stage.datacite.org/contacts/e4056566-7d65-43d6-8efc-d0844f252300
  
<img width="1289" alt="Bildschirmfoto 2021-03-29 um 18 33 44" src="https://user-images.githubusercontent.com/6583633/112869416-6d2fcd00-90bd-11eb-8dac-08b0dc2eb9bf.png">
<img width="1411" alt="Bildschirmfoto 2021-03-29 um 18 33 26" src="https://user-images.githubusercontent.com/6583633/112869429-715bea80-90bd-11eb-9183-20f920481830.png">

**Comment by lizkrznarich** [comment]
Contact deleted from Fabrica still appears in Salesforce 
Contact with email liz.krznarich@datacite.org deleted from organization https://doi.stage.datacite.org/providers/datacite 
Salesforce contact https://datacite--sync.lightning.force.com/lightning/r/Contact/0039E00001CWRwBQAX/view

Fabrica contacts for organization (liz.krznarich@datacite.org does not exist here)
<img width="710" alt="Screen Shot 2021-03-29 at 3 24 04 PM" src="https://user-images.githubusercontent.com/8026886/112895490-d1916300-90a2-11eb-9109-4e83161b6ba5.png">

Corresponding Salesforce organization record
<img width="837" alt="Screen Shot 2021-03-29 at 3 25 00 PM" src="https://user-images.githubusercontent.com/8026886/112895644-fdace400-90a2-11eb-8982-fec765b9e3b1.png">

**Comment by paulvierkant** [comment]
When I renamed Porky Pig to Porky Vierkant in the[ contact details](https://doi.stage.datacite.org/contacts/e4056566-7d65-43d6-8efc-d0844f252300) it did not update in the contact information section of the organization. I had to re-open the dropdown menu to choose Porky Vierkant (see voting representative in the screenshot compared to Porky Pig in the Service Contact). I previous reload of the page did not help.

<img width="1510" alt="Bildschirmfoto 2021-03-30 um 12 46 55" src="https://user-images.githubusercontent.com/6583633/112977153-04e0fa00-9156-11eb-932c-f151bc1608eb.png">
